webapp/channels/src/components/settings_sidebar/settings_sidebar.tsx

```diff
@@ -57,42 +59,72 @@ export default class SettingsSidebar extends React.PureComponent<Props> {
-            if (isActive) {
-                className = 'active';
-            }
+    private renderTab(tab: Tab, index: number) {
+        const key = `${tab.name}_li`;
+        const isActive = this.props.activeTab === tab.name;
+        let className = '';
+        if (isActive) {
+            className = 'active';
+        }
+
+        let icon;
+        if (typeof tab.icon === 'string') {
+            icon = (
+                <i
+                    className={tab.icon}
+                    title={tab.iconTitle}
+                />
+            );
+        } else {
+            icon = (
+                <img
+                    src={tab.icon.url}
+                    alt={tab.iconTitle}
+                    className='icon'
+                />
+            );
+        }
 
-            return (
-                <li
-                    id={`${tab.name}Li`}
-                    key={key}
-                    className={className}
-                    role='presentation'
+        return (
+            <li
+                id={`${tab.name}Li`}
+                key={key}
+                className={className}
+                role='presentation'
+            >
+                <button
+                    ref={this.buttonRefs[index]}
+                    id={`${tab.name}Button`}
+                    className='cursor--pointer style--none'
+                    onClick={this.handleClick.bind(null, tab)}
+                    onKeyUp={this.handleKeyUp.bind(null, index)}
+                    aria-label={tab.uiName.toLowerCase()}
+                    role='tab'
+                    aria-selected={isActive}
+                    tabIndex={!isActive && !this.props.isMobileView ? -1 : 0}
                 >
-                    <button
-                        ref={this.buttonRefs[index]}
-                        id={`${tab.name}Button`}
-                        className='cursor--pointer style--none'
-                        onClick={this.handleClick.bind(null, tab)}
-                        onKeyUp={this.handleKeyUp.bind(null, index)}
-                        aria-label={tab.uiName.toLowerCase()}
-                        role='tab'
-                        aria-selected={isActive}
-                        tabIndex={!isActive && !this.props.isMobileView ? -1 : 0}
-                    >
-                        <i
-                            className={tab.icon}
-                            title={tab.iconTitle}
-                        />
-                        {tab.uiName}
-                    </button>
-                </li>
+                    {icon}
+                    {tab.uiName}
+                </button>
+            </li>
+        );
+    }
+
+    public render() {
+        const tabList = this.props.tabs.map((tab, index) => this.renderTab(tab, index));
+        let pluginTabList: React.ReactNode;
+        if (this.props.pluginTabs.length) {
+            pluginTabList = (
+                <>
+                    <div/>
```

Comment:
I forgot this `div` was supposed to be something like `<div className="divider"/>`. That is the way we general run dividers. We probably should use `<hr>` instead of `<div>`? Probably. But for now, I will try to follow the conventions, unless you have a strong opinion about it.